ttnn/cpp/ttnn/operations/ccl/all_gather/device/kernels/dataflow/worker_interleaved_ring_gather_send_writer.cpp

```diff
@@ -25,48 +45,32 @@ void kernel_main() {
     arg_idx += output_shard_grid_ncols;
     #endif
 
-    constexpr bool dst_is_dram = get_compile_time_arg_val(0) == 1;
-    constexpr uint32_t num_transfers = get_compile_time_arg_val(1);
-    constexpr uint32_t num_full_chunks = get_compile_time_arg_val(2);
-    constexpr uint32_t page_size = get_compile_time_arg_val(3);
-    constexpr uint32_t output_page_size = get_compile_time_arg_val(4);
-    constexpr uint32_t num_pages_per_full_chunk = get_compile_time_arg_val(5);
-    constexpr uint32_t rem_num_pages = get_compile_time_arg_val(6);
-    constexpr uint32_t input_start_idx = get_compile_time_arg_val(7);
-    constexpr uint32_t output_start_idx = get_compile_time_arg_val(8);
-    constexpr uint32_t output_start_addr_offset = get_compile_time_arg_val(9);
-    constexpr uint32_t row_start_idx = get_compile_time_arg_val(10);
-    constexpr uint32_t col_start_idx = get_compile_time_arg_val(11);
-    constexpr uint32_t row_offset = get_compile_time_arg_val(12);
-    constexpr uint32_t col_offset = get_compile_time_arg_val(13);
-    constexpr uint32_t num_rows = get_compile_time_arg_val(14);
-    constexpr uint32_t num_cols = get_compile_time_arg_val(15);
-    constexpr uint32_t input_start_ring_idx = get_compile_time_arg_val(16);
-    volatile uint32_t *const writer_send_sem_ptr = reinterpret_cast<volatile uint32_t *const >(get_semaphore(get_compile_time_arg_val(17)));
-    constexpr uint32_t eth_sender_noc_x = get_compile_time_arg_val(18);
-    constexpr uint32_t eth_sender_noc_y = get_compile_time_arg_val(19);
-    constexpr uint32_t half_cb_n_pages = get_compile_time_arg_val(20);
-    constexpr uint32_t num_buffers_per_channel = get_compile_time_arg_val(21);
-    constexpr bool fuse_op = get_compile_time_arg_val(22);
-
     /* Args for overlapped all gather */
     OpSignaler op_signaler;
 
-    if constexpr(fuse_op) {
+    if (fuse_op) {
```

Comment:
I believe `fuse_op` should be able to remain as a compile time arg? @avoraTT please correct me if I'm wrong.

Reply:
I can't make it compile time arg because this specific one depends on the direction, which is different by cores